docs/pages/database-access/guides/aws-opensearch.mdx

```diff
@@ -0,0 +1,349 @@
+  must be adjusted accordingly for custom configurations.
+
+- (!docs/pages/includes/tctl.mdx!)
+
+<Admonition type="note" title="Example Access Control">
+This guide provides an example configuration of IAM access roles as a model,
+and uses an EC2 instance to serve the Teleport Database Service.
+The level of access provided may not suit your needs, or may not fit your
+organization's access conventions. You should adjust the AWS IAM permissions to fit your needs.
+</Admonition>
+
+## Step 1/4. Create IAM roles for OpenSearch Mange Cluster access
+
+The setup described in this guide requires two IAM roles:
+- One associated with the EC2 instance running the Teleport Database Service,
+  which lets it assume additional roles granted to the user.
+- One that can be assumed by the EC2 instance role and grants access to OpenSearch
+  manage cluster to users.
+
+### EC2 instance role
+
+Visit the [IAM > Roles page](https://console.aws.amazon.com/iamv2/home#/roles) of
+the AWS Console, then press "Create Role". Under **Trusted entity type** select
+"AWS service". Under **Use case** select "EC2", then click **Next**.
+
+![Create Role to Identify EC2 Instance](../../../img/database-access/guides/aws-opensearch/create-ec2-role.png)
+
+On the "Add Permissions" page, you can simply click **Next** since this role does not require any permissions. In this guide, we will use the example name `TeleportDatabaseService` for this role. Once you have chosen a name, click **Create Role** to complete the process.
+
+### OpenSearch Mange Cluster access role
+
+Navigate back to the Roles page and create a new role. Select the "AWS account"
+option, which creates a default trust policy to allow other entities in this
+account to assume this role:
+
+![Create Role Step 1](../../../img/database-access/guides/aws-opensearch/create-role-1.png)
+
+Click **Next**. On the next page, enter a role name. In this guide we'll use
+the example name `ExampleTeleportOpenSearchRole` for this role.
+
+Under "Select trusted entities", update the JSON to allow the `TeleportDatabaseService`
+role to assume this role:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": [
+                    "arn:aws:iam::(=aws.aws_access_key=):role/TeleportDatabaseService"
+                ]
+            },
+            "Action": "sts:AssumeRole",
+            "Condition": {}
+        }
+    ]
+}
+```
+
+Finally, click **Create Role**.
+
+### Configure Cluster Fine-grained access control IAM Role mapping in Amazon OpenSearch Managed Custer.
+
+Teleport AWS OpenSearch service integration leverages the [OpenSearch Fine-grained access control](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/fgac.html) where the IAM role or user is mapped to the OpenSearch role.
+
+In order to configure Role Mapping log into OpenSearch Domain Dashboard an go to the `Security` settings:
```

Comment:
nit: maybe mention login as master/admin user.

To the honest, i think the most difficult part is to get the dashboard working (i run into the anonymous request problem). 

Also, what if opensearch is "VPC" access instead of public access? I haven't tried but i think we can use Teleport app access for the dashboard

Reply:
I have added a reference to master user. Regarding VPC and dashboard access I think that we should not cover this in this docs